rewriter/Struct.cc

```diff
@@ -112,14 +112,32 @@ vector<ast::ExpressionPtr> Struct::run(core::MutableContext ctx, ast::Assign *as
         }
     }
 
+    UnorderedSet<core::NameRef> seenNames;
     for (auto &arg : send->posArgs()) {
-        auto sym = ast::cast_tree<ast::Literal>(arg);
-        if (!sym || !sym->isSymbol()) {
+        auto lit = ast::cast_tree<ast::Literal>(arg);
+        if (!lit) {
             return empty;
         }
-        core::NameRef name = sym->asSymbol();
-        auto symLoc = sym->loc;
-        auto strname = name.shortName(ctx);
+
+        std::string strname;
+        if (lit->isSymbol()) {
+            strname = lit->asSymbol().show(ctx);
+        } else if (lit->isString()) {
+            strname = lit->asString().show(ctx);
+        } else {
+            return empty;
+        }
+
+        core::NameRef name = ctx.state.enterNameUTF8(strname);
+
+        if (seenNames.find(name) != seenNames.end()) {
+            if (auto e = ctx.beginError(lit->loc, core::errors::Rewriter::InvalidStructMember)) {
+                e.setHeader("Duplicate member '{}' in Struct definition", name.show(ctx));
+            }
+            return empty;
+        }
+        seenNames.insert(name);
```

Comment:
It would be better to do:

```c++
auto [_it, inserted] = seenNames.insert(name);
if (!inserted) {
    if (auto e = ctx.beginError(...)) {
        ...
    }
}
```

so that you're only looking up names in the set a single time.